core/src/main/java/com/sequenceiq/cloudbreak/service/filesystem/FileSystemConfigService.java

```diff
@@ -0,0 +1,93 @@
+package com.sequenceiq.cloudbreak.service.filesystem;
+
+import java.util.Set;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.common.model.user.IdentityUser;
+import com.sequenceiq.cloudbreak.common.model.user.IdentityUserRole;
+import com.sequenceiq.cloudbreak.domain.FileSystem;
+import com.sequenceiq.cloudbreak.repository.FileSystemRepository;
+import com.sequenceiq.cloudbreak.service.AuthorizationService;
+
+@Service
+public class FileSystemConfigService {
+
+    @Inject
+    private FileSystemRepository fileSystemRepository;
+
+    @Inject
+    private AuthorizationService authService;
+
+    public Set<FileSystem> retrievePrivateFileSystems(@Nonnull IdentityUser user) {
+        return fileSystemRepository.findForUser(user.getUserId());
+    }
+
+    public FileSystem modify(@Nonnull IdentityUser user, @Nonnull FileSystem fileSystem) {
+        setUserDataRelatedFields(user, fileSystem);
+        authService.hasReadPermission(fileSystem);
```

Comment:
Modification is a write operation, so would be better to check write permissions. Otherwise the create also checks it